### PR TITLE
Add admin users

### PR DESCRIPTION
**Pull request by YummYume**
## Description

#42 #53

## Type of change

Please add any relevant information to the type of change you've made.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [ ] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: front/app/lib/utils/security.server.ts [hunk 1]
```diff
@@ -19,3 +19,12 @@ export const denyAccessUnlessGranted = (
...
 };
+
+/**
+ * Throw a redirect if the user is not an admin. Will return the user if they are an admin.
+ */
+export const denyAdminAccessUnlessGranted = (
+  user: MeUser | null,
+  redirectTo = '/dashboard',
+  init: number | ResponseInit | undefined = undefined,
+) => denyAccessUnlessGranted(user, 'ROLE_ADMIN', redirectTo, init);
```
Comment (Spacelocust) [inline front/app/lib/utils/security.server.ts:30]: Utiliser la constante pour ROLE_ADMIN nn ?

File: front/app/lib/utils/security.server.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export const denyAccessUnlessGranted = (
   roles: string | string[] = 'ROLE_USER',
```
Comment (Spacelocust) [inline front/app/lib/utils/security.server.ts:10]: Pareil pour le ROLE_USER

File: front/app/lib/validators/admin/user.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  balance: zu.number(
+    z
+      .number()
+      .min(Number.MIN_SAFE_INTEGER, { message: 'user.balance.min' })
+      .max(Number.MAX_SAFE_INTEGER, { message: 'user.balance.max' }),
+    {
+      required_error: 'user.balance.required',
+    },
+  ),
+  // locale: z.nativeEnum(UserLocaleEnum, {
```
Comment (Spacelocust) [inline front/app/lib/validators/admin/user.ts:23]: oublie ?

File: front/app/lib/utils/security.server.ts (hunk 1 above)
Comment (YummYume) [reply front/app/lib/utils/security.server.ts:30]: J'ai pas vu de constante dans le front

File: front/app/lib/validators/admin/user.ts (hunk 3 above)
Comment (YummYume) [reply front/app/lib/validators/admin/user.ts:23]: J'attends le select pour mettre

File: front/app/lib/utils/security.server.ts (hunk 1 above)
Comment (Spacelocust) [reply front/app/lib/utils/security.server.ts:30]: Effectivement, je viens de vérifier, je pensais que l'on avait merge ça, mais les constantes sont encore sur la branche "create-heist"

**Review by Spacelocust — DISMISSED** [review]

**Review by Spacelocust — APPROVED** [review]

**Pull request merged**